src/hdf5_summary.F90

```diff
@@ -97,784 +95,659 @@ end subroutine hdf5_write_header
 ! HDF5_WRITE_GEOMETRY
 !===============================================================================
 
-  subroutine hdf5_write_geometry()
+  subroutine hdf5_write_geometry(file_id)
+    integer(HID_T), intent(in) :: file_id
 
     integer          :: i, j, k, m
     integer, allocatable :: lattice_universes(:,:,:)
-    type(Cell),     pointer :: c => null()
-    type(Surface),  pointer :: s => null()
-    type(Universe), pointer :: u => null()
-    class(Lattice), pointer :: lat => null()
+    integer(HID_T) :: geom_group
+    integer(HID_T) :: cells_group, cell_group
+    integer(HID_T) :: surfaces_group, surface_group
+    integer(HID_T) :: universes_group, univ_group
+    integer(HID_T) :: lattices_group, lattice_group
+    type(Cell),     pointer :: c
+    type(Surface),  pointer :: s
+    type(Universe), pointer :: u
+    class(Lattice), pointer :: lat
 
     ! Use H5LT interface to write number of geometry objects
-    call su % write_data(n_cells, "n_cells", group="geometry")
-    call su % write_data(n_surfaces, "n_surfaces", group="geometry")
-    call su % write_data(n_universes, "n_universes", group="geometry")
-    call su % write_data(n_lattices, "n_lattices", group="geometry")
+    geom_group = create_group(file_id, "geometry")
+    call write_dataset(geom_group, "n_cells", n_cells)
+    call write_dataset(geom_group, "n_surfaces", n_surfaces)
+    call write_dataset(geom_group, "n_universes", n_universes)
+    call write_dataset(geom_group, "n_lattices", n_lattices)
 
     ! ==========================================================================
     ! WRITE INFORMATION ON CELLS
 
     ! Create a cell group (nothing directly written in this group) then close
-    call su % open_group("geometry/cells")
-    call su % close_group()
+    cells_group = create_group(geom_group, "cells")
 
     ! Write information on each cell
     CELL_LOOP: do i = 1, n_cells
       c => cells(i)
+      cell_group = create_group(cells_group, "cell " // trim(to_str(c%id)))
 
       ! Write internal OpenMC index for this cell
-      call su % write_data(i, "index", &
-           group="geometry/cells/cell " // trim(to_str(c % id)))
+      call write_dataset(cell_group, "index", i)
 
       ! Write name for this cell
-      call su % write_data(c % name, "name", &
-           group="geometry/cells/cell " // trim(to_str(c % id)))
+      call write_dataset(cell_group, "name", c%name)
```

Comment:
I'm personally in favor of the whitespaces.  Something like `c%name` looks kind of cluttered...  I wish Fortran used a `.` like everybody else.